scripts/check-license-header.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+set -euo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+test -n "${PROJECT_NAME:-}" || fatal "PROJECT_NAME unset"
+
+expected_file_header_template="@@===----------------------------------------------------------------------===@@
+@@
+@@ This source file is part of the ${PROJECT_NAME} open source project
+@@
+@@ Copyright (c) YEARS Apple Inc. and the ${PROJECT_NAME} project authors
+@@ Licensed under Apache License v2.0
+@@
+@@ See LICENSE.txt for license information
+@@ See CONTRIBUTORS.txt for the list of ${PROJECT_NAME} project authors
+@@
+@@ SPDX-License-Identifier: Apache-2.0
+@@
+@@===----------------------------------------------------------------------===@@"
+
+paths_with_missing_license=( )
+
+file_paths=$(git ls-files \
+  ":(exclude).gitignore" \
+  ":(exclude)**/.gitignore" \
+  ":(exclude).licenseignore" \
+  ":(exclude).gitattributes" \
+  ":(exclude).mailfilter" \
+  ":(exclude).mailmap" \
+  ":(exclude).spi.yml" \
+  ":(exclude).swift-format" \
+  ":(exclude).github/*" \
+  ":(exclude)*.md" \
+  ":(exclude)*.txt" \
+  ":(exclude)*.yaml" \
+  ":(exclude)*.json" \
+  ":(exclude)Package.swift" \
+  ":(exclude)**/Package.swift" \
+  ":(exclude)Package.resolved" \
+  ":(exclude)**/Package.resolved" \
+  ":(exclude)Makefile" \
+  ":(exclude)*.modulemap" \
+  ":(exclude)**/*.modulemap" \
+  ":(exclude)**/*.docc/*" \
+  ":(exclude)*.xcprivacy" \
+  ":(exclude)**/*.xcprivacy" \
+  ":(exclude)*.symlink" \
+  ":(exclude)**/*.symlink" \
+  ":(exclude)Dockerfile" \
+  ":(exclude)**/Dockerfile" \
+  ":(exclude)Snippets/*" \
+  $(printf ":(exclude)%s " $(cat .licenseignore))
+)
+
+while IFS= read -r file_path; do
+  file_basename=$(basename -- "${file_path}")
+  file_extension="${file_basename##*.}"
+
+  case "${file_extension}" in
+    swift) expected_file_header=$(sed -e 's|@@|//|g' <<<"${expected_file_header_template}") ;;
+    h) expected_file_header=$(sed -e 's|@@|//|g' <<<"${expected_file_header_template}") ;;
+    c) expected_file_header=$(sed -e 's|@@|//|g' <<<"${expected_file_header_template}") ;;
+    yml) expected_file_header=$(sed -e 's|@@|##|g' <<<"${expected_file_header_template}") ;;
```

Comment:
Can you add `yaml` as well?

Reply:
We are actually ignoring all `yml` and `yaml` files so we don't have to check them.